### PR TITLE
fix dropouts for future hypernetworks

### PR DESCRIPTION
**Pull request by aria1th**
The line from https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3414/commits/6a4fa73a38935a18779ce1809892730fd1572bee#diff-bf73e9cd868899b3905c9ec9e2af2d9cda9f69cd3a52c27e5584382bcb281ad2R55 caused missing last 2nd(slice -2) dropout layer.

Although the effect of dropout is controversial, for consistency, dropout layers might have to be there.

This pr solves the problem of loading previous ones, which does not have 'last_layer_dropout' key in state_dict -> defaulted to False, so it matches previous structures.

Newly created networks will apply dropout at -2 slice, or, user can choose to skip [-2] slice dropout like before. The option activated to less or equal than 3 layers networks, will result dropout less networks.

Though I doubt the usage of more than 2 dropouts with high probability(like current 0.3), rather, it should be replaced with sophisticated controllable parameters - layer based control if possible.

https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3698

### CODE REVIEW

**Comment by guaneec** [comment]
I fixed a logic error where no dropout would be added if `last_layer_dropout` is False. Now old nets should load properly.

I think we should just silently apply the last layer dropout fix without adding a new checkbox, as that might just be confusing to users. Old HNs aren't affected anyway, and I doubt that people would actually prefer new HNs created with a missing dropout layer.

**Comment by aria1th** [comment]
I agree, what people might really need is controllable parameters, not checkbox for 'what is this'.
I'll remove the option within 6 hours... or you can simply revert changes in ui.

**Comment by aria1th** [comment]
removed option and squashed commits
